scripts/optimize.py

```diff
@@ -81,8 +92,9 @@ def get_emission_limit():
         m = Model(es)
 
         # Add an emission constraint
-        if emission_limit is not None:
-            constraints.emission_limit(m, limit=emission_limit)
+        if scalars is not None:
```

Comment:
You can omit checking this because you have checked it above already.

Reply:
I need to check it here again, as `emission_limit` does not exist if `scalars` is None. See here: https://github.com/rl-institut/oemof-B3/blob/e8ca7ab67a4ddf9a6b339d6b422817f0ac1b1689/scripts/optimize.py#L78-L79

Another option would be to add `else` to L78-L79 
```
    if additional_scalars is not None:
        emission_limit = get_emission_limit(additional_scalars)
    else:
        emission_limit = None

```

This will get quite long, as other variables will have to be set here, too, see #97.
This is why I've moved that into a function in the first place.

I would just keep the `if scalars is not None:` here a second time to save lines and for easier understanding.

Reply:
Proposal: Set `emission_limit = None` at the very begging.